### ISSUE TITLE
Issue with cell background color in 8.1

### ISSUE THREAD

**Issue by Neonsly** [PHPOffice/PhpSpreadsheet#2528]
This is:

```
- [X ] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
`setARGB('FFA07A');` - set white color instead of FFA07A

### What is the current behavior?
Only few color is working  instead of start and end color ARGB settings.
This works 
`setARGB(\PhpOffice\PhpSpreadsheet\Style\Color::COLOR_RED);`
This not working 
`setARGB('FFA07A');`

### What are the steps to reproduce?
Upgrade from 7.3 to 8.1 php
I think working only colors from documenation 

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

// Create new Spreadsheet object
$spreadsheet = new Spreadsheet();								
$sheet = $spreadsheet->getActiveSheet();

// add code that show the issue here...

$spreadsheet->getActiveSheet()->getStyle('A1')->getFill()->setFillType(\PhpOffice\PhpSpreadsheet\Style\Fill::FILL_SOLID)->getStartColor()->setARGB('FFA07A');	

$styleArray_header_grad = [
		'font' => [
			'bold' => true,
		],
		'alignment' => [
			'horizontal' => \PhpOffice\PhpSpreadsheet\Style\Alignment::HORIZONTAL_CENTER,
		],
		'borders' => array(
			'outline' => array(
				'borderStyle' => \PhpOffice\PhpSpreadsheet\Style\Border::BORDER_THIN,
				'color' => array('argb' => '000000'),
			)
		),			    
		'fill' => [
			'fillType' => \PhpOffice\PhpSpreadsheet\Style\Fill::FILL_GRADIENT_LINEAR,
			'rotation' => 90,
			'startColor' => [
				'argb' => 'FFA0A0A0',
			],
			'endColor' => [
				'argb' => 'FFFFFFFF',
			],
		],
	];
$sheet->getStyle('B2:E2')->applyFromArray($styleArray_header_grad );	

$writer = new Xlsx($spreadsheet);
$writer->save('../resources/text.xlsx');

### Which versions of PhpSpreadsheet and PHP are affected?
1.21 | 8.1

**Comment by Neonsly** [comment]
[](url)

**Comment by Neonsly** [comment]
In new version - ARGB with 6 numbers hex code not working, need to use RGB

**Comment by BARNZ** [comment]
I'm getting this issue today as well - happens on php7.4 as well.

I previously had the color `C5D9F1` which used to work, but now I need to specify `FFC5D9F1 (FF at the front).

**Comment by andrec93p** [comment]
> I'm trying to make a gray background, but any shade I use turns into a yellow background for some reason.

Ok, I figured out a solution literally seconds after submitting this.

This code apparently works.
```
$activeSheet
    ->getStyle('A' . $rowIndex . ':' . $columnIndex . $rowIndex)
    ->applyFromArray([
        'font' => [
            'bold' => true,
        ],
        'fill' => [
            'fillType' => \PhpOffice\PhpSpreadsheet\Style\Fill::FILL_SOLID,
            'startColor' => [
                // Color 15 on the standard palette is #FFC0C0C0.
                // Found in the Color class, inside the INDEXED_COLORS array const.
                'argb' => \PhpOffice\PhpSpreadsheet\Style\Color::indexedColor(15)->getARGB(),
            ],
        ],
    ])
;
```
Still, this looks sort of weird to me. Is this intended behaviour? Is there any technical reason why I can't use any valid ARGB code?

**Comment by oleibman** [comment]
@andrec93p The following code works for me (I changed F7F7F7 to C0C0C0 for better visibility, but the result is essentially the same in either case).
```php

require __DIR__ . '/PhpSpreadsheet' . '/vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Style\Color;
use PhpOffice\PhpSpreadsheet\Style\Fill;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx as XlsxWriter;

class issue2528 {
    public static function issue() {
        $spreadsheet = new Spreadsheet();
        $sheet = $spreadsheet->getActiveSheet();
        $sheet->fromArray([['A', 'B', 'C'], [1, 2, 3], [4,5,6]]);
        $sheet
            ->getStyle('A1:C1')
            ->applyFromArray([
                'font' => [
                    'bold' => true,
                ],
                'fill' => [
                    'fillType' => Fill::FILL_SOLID,
                    'startColor' => [
                        //'argb' => Color::COLOR_BLUE, // works
                        'argb' => 'FFC0C0C0', // doesn't work
                    ],
                ],
            ])
        ;
        $writer = new XlsxWriter($spreadsheet);
        $outfile = 'issue.2528.xlsx';
        $writer->save($outfile);
        echo "saved $outfile\n";
    }
```
![image](https://github.com/user-attachments/assets/81100464-0ad0-4bae-b5bb-d8962e5d1f98)

So I think it is working with any valid ARGB argument. I do not yet have a theory for what's happening in your environment.

**Comment by oleibman** [comment]
BTW, you can now enter the ARGB color, at least in this example, as either 8-hex-digit ARGB or 6-hex-digit RGB (A will be treated as FF in that case).

**Comment by andrec93p** [comment]
@oleibman hey, thanks a lot, I was not expecting such a quick reply.

I tried your example, and it does work with both `FFC0C0C0` (which is a standard color listed in the `Color` class by the way) and `FFF7F7F7` (non-standard). Something is definitely wrong in my environment.

I'm using the library in Symfony v6.4. I tried creating a fresh Symfony project to see if I was able to reproduce the issue, but your code was working there too. I really didn't understand what was going on... except now the code works on my actual project as well. I have no idea what happened yesterday, so don't mind my posts.

Sorry for bothering you, thanks again for the help.
